pgb_utils/tutorials/PGB_Tutorial.ipynb

```diff
@@ -0,0 +1,2756 @@
+      "metadata": {
+        "id": "_vE16R7x4yJB"
+      },
+      "source": [
+        "The resources used in completing this tutorial will remain well within Google Cloud's [Free Tier](https://cloud.google.com/free)(*).\n",
+        "If you wish to become a power-user in the future, you may need to [create a billing account](https://support.google.com/cloud/answer/6293499#enable-billing). See also: \n",
+        "- [pricing structure](https://cloud.google.com/pricing) (scroll to \"Only pay for what you use\")\n",
+        "- [detailed price list](https://cloud.google.com/pricing/list) (search for \"BigQuery\", \"Cloud Storage\", \"Pub/Sub\"); \n",
+        "- [pricing calculator](https://cloud.google.com/products/calculator?skip_cache=true) (same search as above)\n",
+        "\n",
+        "(*) The Free Tier for Cloud Storage is limited to US regions. Please see that section of this tutorial for details. I am interested in feedback from users outside the US on the behavior of that section of this tutorial for them."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ALIGt4P12KB-"
+      },
+      "source": [
+        "---"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "pdVlJmWWawvx"
+      },
+      "source": [
+        "# 1) Setup"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3qjOE_7Ka1zR"
+      },
+      "source": [
+        "\n",
+        "1. Use the [Cloud Resource Manager](https://console.cloud.google.com/cloud-resource-manager) to create a GCP project. Take note of the auto-generated \"Project ID\"; you will need it below.\n",
+        "    - All our data is public and hosted by [Google Cloud](https://cloud.google.com/). In order to make API calls you will need to create a Cloud project that is associated with your Google account.\n",
+        "    - If you work outside of Colab you may need to:\n",
+        "        - enable APIs for your project: [BigQuery](https://console.cloud.google.com/flows/enableapi?apiid=bigquery); \n",
+        "        - [setup authentication](https://cloud.google.com/docs/authentication/getting-started)\n",
+        "\n",
+        "2. Installs, imports, etc.:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "qXFXbjlVYFvZ"
+      },
+      "source": [
+        "from google.colab import auth"
+      ],
+      "execution_count": 9,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hm8S4HbIQlK1"
+      },
+      "source": [
+        "auth.authenticate_user()\n",
+        "# follow the instructions to authorize Google Cloud SDK "
+      ],
+      "execution_count": 10,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ZwISaJSymrhh"
+      },
+      "source": [
+        "pgb_project_id = 'ardent-cycling-243415'"
```

Comment:
Not without writing/using some mapping function. All of the calls that use it will only take the project ID.
I could add a line that defines our project name if it reduces confusion, but it wouldn't actually be used anywhere.

```python
pgb_project_name = 'pitt-google-broker-prototype'
pgb_project_id = 'ardent-cycling-243415'
```

Reply:
Ah, I misunderstood.  I though you could you use either.  But now I understand that the project_id must be unique by the project_name likely doesn't have to be.  The project_name is just a useful label for your own use.  Thanks.

https://cloud.google.com/resource-manager/docs/creating-managing-projects